docs/DeveloperGuide.md

```diff
@@ -270,51 +270,189 @@ _{Explain here how the data archiving feature will be implemented}_
+    * 1a1. JerylFypManager shows an error message
+    * 1a2. Professor keys in his input again
+
+    Steps 1a1-1a2 are repeated until the request entered is correct
+
+  Use case resumes from step 2
+
+
+**Use case: UC04 - Removing FYP**
+
+**Preconditions: Professor can recall the name/id of the FYP**
+
+**MSS**
+
+1.  Professor requests to remove a student FYP into the list
+2.  JerylFypManager removes the FYP details
+
+    Use case ends
+
+**Extensions**
+
+* 1a. The request given is invalid
+    * 1a1. JerylFypManager shows an error message
+    * 1a2. Professor enters a new request
+
+  Steps 1a1-1a2 are repeated until the request entered is correct
+
+  Use case resumes from step 2
+
+
+**Use case: UC05 - Help List**
+
+**Preconditions: Nil**
+
+**MSS**
+
+1.  Professor requests from JerylFypManager to list all possible commands
+2.  JerylFypManager shows the list of possible commands and format to be entered in
+
+    Use case ends
+
+**Extensions**
+
+* Nil
+
+
+**Use case: UC06 - List FYPs**
+
+**Preconditions: There exist FYP(s) in JerylFypManager**
+
+**MSS**
+
+1.  Professor requests for JerylFypManager to show the list of FYPs
+2.  JerylFypManager shows a list of FYPs to the professor
+
+    Use case ends
+
+
+**Extensions**
+
+* 2a. There are currently no FYPs that the professor is overseeing
+    * 2a1. JerylFypManager tells the professor that he is currently not overseeing any FYPs
+
+  Use case ends
+
+
+**Use case: UC07 - Exit JerylFypManager**
+
+**Preconditions: Nil**
+
+**MSS**
+
+1.  Professor requests to exit JerylFypManager
+2.  JerylFypManager shows exit message
+3.  JerylFypManager application is closed
+
+    Use case ends
+
+**Extensions**
 
-    * 3a1. AddressBook shows an error message.
+* Nil
 
-      Use case resumes at step 2.
 
 *{More to be added}*
 
 ### Non-Functional Requirements
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+2.  User with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+3.  As a busy SoC professor, I want to be able to see my students’ project titles and summaries at one glance, so that I do not have to waste time clicking into different links to access them.
+4.  As an insecure FYP student, I have to be able to ring or notify my professor(s) for help via the platform, so that I get the immediate help that I require.
+5.  As a fun-seeking professor/FYP student , I want to be able to send “stickers” (similar to Telegram stickers) or emoji, so that the app is more appealing to use.
```

Comment:
Do you think it's better to rephrase this such that it doesn't like it's just a raw copy-paste from the user stories list?

Reply:
Should we make it more generic? Refering to the examples in point 1 and 2, should we make it more generic instead of  a specific target auidence?

Reply:
Actually thinking about it points 4 and 5 seems to be a functional requirement. How about changing to something like these instead?
```
3. The system should respond in at most 1 second.
4. A user with no coding background should be able to use JerylFypManager.
```

Reply:
That sounds alot better. Thanks for the suggestion boss